Controllers/AbstractControllers/AbstractUserController.php

```diff
@@ -4,65 +4,91 @@
-use Models\AbstractProjectModels\Message\User\AbstractMsgModel;
-use Models\ProjectModels\Logger;
+//use mysql_xdevapi\Exception;
 
 abstract class AbstractUserController extends AbstractBaseController
 {
+    protected const REQUEST = 'request';
+    protected const REFERER = 'referer';
+    protected const CUSTOMER = 'customer';
+    protected const CONTROLLER = 'controller';
+    protected const ACTION = 'action';
+    protected const EMPTY = 'empty';
+    protected const WRONG = 'wrong';
+    protected const IMAGE = 'image';
+
     protected AbstractUserModel $userModel;
     protected AbstractDefaultView $userView;
-    protected Validator $dataValidator;
-    protected AbstractMsgModel $msgModel;
+    protected ?IDataManagement $serverInfo = null;
+    protected ?UserDataValidatorInterface $dataValidator = null;
+    protected ?AbstractBaseMsgModel $msgModel = null;
     protected ?ImageValidator $imageValidator = null;
     protected ?IDataManagement $postInfo = null;
     protected ?IDataManagement $fileInfo = null;
-    protected Logger $logger;
 
     public function __construct(
         AbstractUserModel $userModel,
-        AbstractDefaultView $userView,
-        Validator $dataValidator,
-        AbstractMsgModel $msgModel
+        AbstractDefaultView $userView
     ) {
         $this->userModel = $userModel;
         $this->userView = $userView;
-        $this->dataValidator = $dataValidator;
-        $this->msgModel = $msgModel;
-        $this->logger = new Logger();
     }
 
     public function registrationAction(array $params = null): void
     {
-        if (!$this->userModel->isSigned()) {
-            $options = $this->userView->getOptions('Регистрация', 'registration.phtml');
-            $this->userView->render($options);
-        } else {
+        if ($this->userModel->isSigned()) {
             $this->redirectHomeByCustomerType();
+
+            return;
+        }
+
+        try {
+            $this->getMsgModel(self::REQUEST);
+            if ($params !== null) {
+                throw new \Exception('Params have to be empty in registrationAction!');
+            }
+
+            $this->userView->render($this->userView->getOptions('Регистрация', 'registration.phtml'));
+        } catch (\Exception $exception) {
+            $this->exceptionCatcher($exception);
         }
     }
 
-    public function authorizationAction(): void
+    public function authorizationAction(array $params = null): void
     {
-        if (!$this->userModel->isSigned()) {
-            $options = $this->userView->getOptions('Авторизация', 'login.phtml');
-            $this->userView->render($options);
-        } else {
+        if ($this->userModel->isSigned()) {
             $this->redirectHomeByCustomerType();
+
+            return;
+        }
+
+        try {
+            $this->getMsgModel(self::REQUEST);
+            if ($params !== null) {
+                throw new \Exception('Params have to be empty in authorizationAction!');
```

Comment:
So you throw the exception here in this function and catch it here as well? What is the purpose of this move?